mikupad.html

```diff
@@ -2184,6 +2194,14 @@
 		}
 	}, [theme]);
 
+
+	useEffect(() => {
+		if (shouldCallPredict) {
+			predict();
+			setShouldCallPredict(false);
+		}
+	}, [promptChunks]); 
+

```

Comment:
`shouldCallPredict` is unnecessary, see `switchCompletion`.
https://github.com/lmg-anon/mikupad/blob/d714989a443144edfd561429e3f9779a252d32aa/mikupad.html#L2458-L2468

Reply:
Let me check. I'm not too familiar with react (my background is C#) and when I tried to call directly predict after undo, I got a DOM error which I guessed was due to both methods updating the DOM concurrently. It's also true that by mistake I had formed from another fork of your repo, and it was way behind. So much I had to start from scratch.

Reply:
No problem, `promptChunks` is a state, and the `setPromptChunks` only affects the next render, that's why in `predict` is being called like that in `switchCompletion`.

Reply:
Looks fine to me now.